src/oemof/solph/components/_generic_storage.py

```diff
@@ -596,7 +606,18 @@ def _objective_expression(self):
                             * ((1 + m.discount_rate) ** -p)
                         )
         self.fixed_costs = Expression(expr=fixed_costs)
-        return self.fixed_costs
+
+        storage_costs = 0
+
+        for n in self.STORAGES:
+            if n.storage_costs[0] is not None:
+                storage_costs += self.storage_content[n, 0] * n.storage_costs[0]
+                for t in m.TIMESTEPS:
+                    storage_costs += self.storage_content[n, t+1] * n.storage_costs[t+1]
```

Comment:
The earlier described behavior probably results somehow from having the cumulated sum of costs over all timsteps. Maybe a note should be added that it might be complicated to directly steer the behavior?!